### ISSUE TITLE
Training questions

### ISSUE THREAD

**Issue by cwhazzoo** [nagadomi/waifu2x#9]
Took me a while to get everything installed and working. Right now I am doing noise training with about 300 images. It is currently on run #53. Just curious if there is a set number of runs it goes through?

**Comment by nagadomi** [comment]
training step runs #200 epochs by default.

**Comment by an-electric-sheep** [comment]
Another question about training: does the `batch_size` make any quality difference or is it just for performance tuning?

**Comment by nagadomi** [comment]
In waifu2x,  computing performance mostly not influenced by batch_size (but slow performance when batch_size=1).
I recommend small batch_size (2~4).
small batch_size: low memory usage. fast convergence. reducing overtting.

**Comment by drdaxxy** [comment]
I have another question: how long did the entire training process take for your training set?

**Comment by nagadomi** [comment]
2x scaling model: 16 hours
Noise reduction model: 28 hours * 2
Total: around 72 hours
(on GTX760, 3000 images)

**Comment by c0d3rman** [comment]
The facts brought up here (how long training takes, batch size recommendation, epochs to training end) should be added to the documentation - they would be EXTREMELY useful to someone like me trying to train their own model.

**Comment by djdjoko** [comment]
Could you please let me know how many iterations did you train it for? Also did you use all the current default settings when training your anime model?

**Comment by nagadomi** [comment]
See https://github.com/nagadomi/waifu2x/blob/master/train.sh
The default setting is defined at https://github.com/nagadomi/waifu2x/blob/master/lib/settings.lua
I was used 3000 images for training.

And I'm currently running the traning script for next version using [dev branch code](https://github.com/nagadomi/waifu2x/tree/dev) and 6000 images.

**Comment by djdjoko** [comment]
Thanks! Could you share what the next version is and what the differences are?

**Comment by nagadomi** [comment]
I released new version yesterday.
Summary of difference in training script:
- Add data compression method for on-memory image (reducing memory usage for dataset)
- Change loss function (MSE to Customized Huber Loss)
- Improve some image processing functions
- Add some data augmentation methods
- Tuning default parameters
- Improve code readability
- Improve training time (4x~ faster)

Sorry, my english skill is very poor. Please refer the code for more details.

**Comment by djdjoko** [comment]
I would like to ask you a hypothetical question. If I have only one specific type of image I want to scale up, would it be possible to train the algorithm only on one perfect image with maximum resolution?
